source/faucet/main.d

```diff
@@ -84,145 +86,172 @@ private struct State
+        A task called periodically that generates and send transactions to a node
 
-    immutable median = sutxo[size / 2].output.value;
-    // Should be 500M (5,000,000,000,000,000) for the time being
-    immutable sum = sutxo.map!(utxo => utxo.output.value)
-        .fold!((a, b) => Amount(a).mustAdd(b))(Amount(0));
-    auto mean = Amount(sum); mean.div(size);
+        This function will wait for block 1 to be externalized before doing anything
+        (block 1 should be triggered by `setup`).
+        Each time this runs, it creates 16 transactions which split an UTXO among
+        15 random keys.
 
-    logInfo("\tMedian: %s, Avg: %s", median, mean);
-    logInfo("\tL: %s, H: %s", sutxo[0].output.value, sutxo[$-1].output.value);
+        Params:
+          client = An API instance to connect to a node
 
-    if (state.utxos.storage.length > 200)
-    {
-        auto tx = state.utxos.byKeyValue().take(16).mergeTx();
-        client.putTransaction(tx);
-        logDebug("Transaction sent: %s", tx);
-    }
-    else
+    *******************************************************************************/
+
+    void send (API client)
     {
-        foreach (tx; state.utxos.byKeyValue().take(16).splitTx(Config.count))
+        this.state.update(client, Height(this.state.known + 1));
+        if (this.state.known < 1)
+            return logInfo("Waiting for setup to be completed");
+
+        logInfo("About to send transactions...");
+
+        // Sort them so we don't iterate multiple time
+        // Note: This may cause a lot of memory usage, might need restructuing later
+        // Mutable because of https://issues.dlang.org/show_bug.cgi?id=9792
+        auto sutxo = this.state.utxos.values.sort!((a, b) => a.output.value < b.output.value);
+        const size = sutxo.length();
+        logInfo("\tUTXO set: %d entries", size);
+
+        immutable median = sutxo[size / 2].output.value;
+        // Should be 500M (5,000,000,000,000,000) for the time being
+        immutable sum = sutxo.map!(utxo => utxo.output.value)
+            .fold!((a, b) => Amount(a).mustAdd(b))(Amount(0));
+        auto mean = Amount(sum); mean.div(size);
+
+        logInfo("\tMedian: %s, Avg: %s", median, mean);
+        logInfo("\tL: %s, H: %s", sutxo[0].output.value, sutxo[$-1].output.value);
+
+        if (this.state.utxos.storage.length > 200)
         {
+            auto tx = this.mergeTx(this.state.utxos.byKeyValue().take(16));
             client.putTransaction(tx);
             logDebug("Transaction sent: %s", tx);
         }
+        else
+        {
+            foreach (tx; this.splitTx(this.state.utxos.byKeyValue().take(16),
+                                      this.config.count))
+            {
+                client.putTransaction(tx);
+                logDebug("Transaction sent: %s", tx);
+            }
+        }
+    }
+
+    /// GET: /utxos
+    public override UTXO[Hash] getUTXOs () pure nothrow @safe
```

Comment:
Could you move this into a separate commit? Because this commit refactors the code and also adds a new feature.